### PR TITLE
Generator refactoring.

### PR DESCRIPTION
**Pull request by winder**
* `Generator` -> `OpenApiParser`
* `legacyMode` flag in `OpenApiParser` so that the parser can run without supplying java configurations.
* `parse()` function in `OpenApiParser` to more intuitively run the operations done by the static `Generate` method (and only the required operations).
* Improved CLI parsing to allow multiple subcommands.
* Stubbed in `responses` subcommand.

### CODE REVIEW

File: generator/src/main/java/com/algorand/sdkutils/Generate.java [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+     * @param modelPackage   Package name to put at the top of generated model files.
+     * @param pathsPath      Path where generated path builder files will be placed.
+     * @param pathsPackage   Package name to put at the top of generated path builder files.
+     * @param commonPath     Path where the common files are, this is where the client class will be placed.
+     * @param commonPackage  Package name to put at the top of generated client class.
+     * @param tokenName      Name of the token used for this application. i.e. X-Algo-API-Token
+     * @param tokenOptional  Whether or not a no-token version of the constructor should be created.
+     * @param goDirectory    When specified, will generate go code.
+     */
+    @Deprecated
```
Comment (algonautshant) [inline generator/src/main/java/com/algorand/sdkutils/Generate.java:30]: We don't need to deprecate. 
If we have the equivalent one, then just delete this, and use OpenApiParser:parse

File: generator/src/main/java/com/algorand/sdkutils/generators/OpenApiParser.java [hunk 2]
```diff
@@ -20,11 +18,40 @@
 import com.fasterxml.jackson.databind.node.ArrayNode;
 import com.algorand.sdkutils.utils.*;
 
-public class Generator {
+public class OpenApiParser {
     public static final String TAB = "    ";
 
     protected JsonNode root;
     protected Publisher publisher;
+    protected final boolean legacyMode;
```
Comment (algonautshant) [inline generator/src/main/java/com/algorand/sdkutils/generators/OpenApiParser.java:26]: let's name it javaGeneration. 'legacy' is not informative and confusing.

File: generator/src/main/java/com/algorand/sdkutils/Generate.java (hunk 1 above)
Comment (winder) [reply generator/src/main/java/com/algorand/sdkutils/Generate.java:30]: It's still needed, there are some extra steps that is only used by the java generator in this one.

The GoGenerator could now migrate pretty easily though.

File: generator/src/main/java/com/algorand/sdkutils/generators/OpenApiParser.java (hunk 2 above)
Comment (winder) [reply generator/src/main/java/com/algorand/sdkutils/generators/OpenApiParser.java:26]: Sounds good, I'll rename it in my other PR.

**Pull request merged**